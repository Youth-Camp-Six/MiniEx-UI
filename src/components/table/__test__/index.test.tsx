/* eslint-disable */
/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Table, TableColumns } from '../index';
import { Avatar, Flex, Badge } from '../../../index'
describe('test Table', () => {
  var data = [
    {
      id: 1,
      name: '小明',
      avatar: 'https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668',
      age: '14',
      fraction: 63,
      mail: 'div@kamtao.com',
      sex: 1,
      intro: '从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。',
    },
    {
      id: 2,
      name: '小卢',
      avatar: 'https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668',
      age: '24',
      fraction: 82,
      mail: 'div@kamtao.com',
      sex: 1,
      intro: '从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。',
    },
    {
      id: 3,
      name: '小娟',
      avatar: 'https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668',
      age: '25',
      fraction: 91,
      mail: 'div@kamtao.com',
      sex: 0,
      intro: '从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。',
    },
    {
      id: 4,
      name: '小贝',
      avatar: 'https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668',
      age: '22',
      mail: 'div@kamtao.com',
      fraction: 31,
      sex: 0,
      intro: '从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。',
    },
    {
      id: 5,
      name: '小飞',
      avatar: 'https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668',
      age: '13',
      fraction: 54,
      mail: 'div@kamtao.com',
      sex: 3,
      intro: '从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。',
    },
  ]
  const columns: TableColumns[] = [
    {
      title: 'id',
      width: '100px',
      field: 'id',
      x: 'center',
    },
    {
      title: '信息',
      width: '180px',
      field: 'info',
    },
    {
      title: '年龄',
      width: 'auto',
      field: 'age',
      x: 'center',
    },
    {
      title: '性别',
      width: 'auto',
      field: 'sex',
      x: 'center',
    },
    {
      title: '成绩',
      width: 'auto',
      field: 'fraction',
      x: 'center',
    },
    {
      title: '介绍',
      width: 'auto',
      field: 'intro',
    },
  ]
  const cell = (field: any, row: any) => {
    switch (field) {
      case "id": return <div>{row.id}</div>
      case "info": return <Flex className="info">
        <Avatar src={row.avatar} round='round'></Avatar>
        <Flex className="right" direction="y" x="start" gap={0}>
          <div className="name">{row.name}</div>
          <div className="mail">{row.mail}</div>
        </Flex>
      </Flex>
      case "age": return <div>{row.age}</div>
      case "fraction": return <Flex>
        <Badge round='round' type={`${row.fraction >= 60 ? "success" : "error"}`} ></Badge>
        {row.fraction >= 60 ? '良好' : '很差'}
      </Flex>
      case "sex": return <div >{row.sex > 0 ? '男' : '女'}</div>
      case "intro": return <div>{row.intro}</div>
    }

  }
  it('Table snapshot test', () => {
    const wrapper = render(<div style={{ width: '813px' }}>
      <Table data={data} columns={columns} cell={cell}></Table>
    </div>
    )
    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

});
