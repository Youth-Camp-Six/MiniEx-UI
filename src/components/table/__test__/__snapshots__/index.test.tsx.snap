// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Table Table snapshot test 1`] = `
<div
  style="width: 813px;"
>
  <div
    class="mi-table"
    style="overflow-x: auto; overflow-y: hidden;"
  >
    <div
      class="mi-table-head"
    >
      <div
        class="mi-table-tr"
      >
        <div
          class="mi-flex mi-table-td 
              
              
              
               mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 100px;"
        >
          id
        </div>
        <div
          class="mi-flex mi-table-td 
              
              
              
               mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 180px;"
        >
          信息
        </div>
        <div
          class="mi-flex mi-table-td 
              
              
              
               mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          年龄
        </div>
        <div
          class="mi-flex mi-table-td 
              
              
              
               mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          性别
        </div>
        <div
          class="mi-flex mi-table-td 
              
              
              
               mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          成绩
        </div>
        <div
          class="mi-flex mi-table-td 
              
              
              
               mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          介绍
        </div>
      </div>
    </div>
    <div
      class="mi-table-body"
    >
      <div
        class="mi-table-tr"
      >
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 100px;"
        >
          <div>
            1
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 180px;"
        >
          <div
            class="mi-flex info mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-avatar"
              style="width: 40px; height: 40px;"
            >
              <img
                alt=""
                class="mi-avatar-round"
                src="https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668"
              />
            </div>
            <div
              class="mi-flex right mi-flex-y mi-flex-x-start mi-flex-y-center"
              style="gap: 0px;"
            >
              <div
                class="name"
              >
                小明
              </div>
              <div
                class="mail"
              >
                div@kamtao.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            14
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            男
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div
            class="mi-flex mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-badge mi-badge-success mi-badge-round"
            >
              <div
                class="mi-badge-dot"
              />
            </div>
            良好
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。
          </div>
        </div>
      </div>
      <div
        class="mi-table-tr"
      >
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 100px;"
        >
          <div>
            2
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 180px;"
        >
          <div
            class="mi-flex info mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-avatar"
              style="width: 40px; height: 40px;"
            >
              <img
                alt=""
                class="mi-avatar-round"
                src="https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668"
              />
            </div>
            <div
              class="mi-flex right mi-flex-y mi-flex-x-start mi-flex-y-center"
              style="gap: 0px;"
            >
              <div
                class="name"
              >
                小卢
              </div>
              <div
                class="mail"
              >
                div@kamtao.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            24
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            男
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div
            class="mi-flex mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-badge mi-badge-success mi-badge-round"
            >
              <div
                class="mi-badge-dot"
              />
            </div>
            良好
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。
          </div>
        </div>
      </div>
      <div
        class="mi-table-tr"
      >
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 100px;"
        >
          <div>
            3
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 180px;"
        >
          <div
            class="mi-flex info mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-avatar"
              style="width: 40px; height: 40px;"
            >
              <img
                alt=""
                class="mi-avatar-round"
                src="https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668"
              />
            </div>
            <div
              class="mi-flex right mi-flex-y mi-flex-x-start mi-flex-y-center"
              style="gap: 0px;"
            >
              <div
                class="name"
              >
                小娟
              </div>
              <div
                class="mail"
              >
                div@kamtao.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            25
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            女
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div
            class="mi-flex mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-badge mi-badge-success mi-badge-round"
            >
              <div
                class="mi-badge-dot"
              />
            </div>
            良好
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。
          </div>
        </div>
      </div>
      <div
        class="mi-table-tr"
      >
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 100px;"
        >
          <div>
            4
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 180px;"
        >
          <div
            class="mi-flex info mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-avatar"
              style="width: 40px; height: 40px;"
            >
              <img
                alt=""
                class="mi-avatar-round"
                src="https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668"
              />
            </div>
            <div
              class="mi-flex right mi-flex-y mi-flex-x-start mi-flex-y-center"
              style="gap: 0px;"
            >
              <div
                class="name"
              >
                小贝
              </div>
              <div
                class="mail"
              >
                div@kamtao.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            22
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            女
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div
            class="mi-flex mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-badge mi-badge-error mi-badge-round"
            >
              <div
                class="mi-badge-dot"
              />
            </div>
            很差
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。
          </div>
        </div>
      </div>
      <div
        class="mi-table-tr"
      >
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 100px;"
        >
          <div>
            5
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: 180px;"
        >
          <div
            class="mi-flex info mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-avatar"
              style="width: 40px; height: 40px;"
            >
              <img
                alt=""
                class="mi-avatar-round"
                src="https://q1.qlogo.cn/g?b=qq&s=100&nk=1057072668"
              />
            </div>
            <div
              class="mi-flex right mi-flex-y mi-flex-x-start mi-flex-y-center"
              style="gap: 0px;"
            >
              <div
                class="name"
              >
                小飞
              </div>
              <div
                class="mail"
              >
                div@kamtao.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            13
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            男
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div
            class="mi-flex mi-flex-x mi-flex-x-center mi-flex-y-center"
            style="gap: 10px;"
          >
            <div
              class="mi-badge mi-badge-error mi-badge-round"
            >
              <div
                class="mi-badge-dot"
              />
            </div>
            很差
          </div>
        </div>
        <div
          class="mi-flex mi-table-td 
                
                
                
                 mi-flex-x mi-flex-x-center mi-flex-y-center"
          style="position: sticky; width: -70.8px;"
        >
          <div>
            从小独立呼吸，讲卫生懂礼貌，不会随便捡地上的东西吃。
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
